### PR TITLE
Stop emitting the features section in some buggy cases

### PR DESCRIPTION
**Pull request by kripken**
`wasm-emscripten-finalize` strips the features section (after emitting metadata using it),
but we no longer always run that tool, so we need to strip it manually otherwise.

This is a longstanding issue that was noticed on 

https://github.com/WebAssembly/binaryen/pull/5996#issuecomment-1751380845

The features section became large enough to be noticed in a test that
used the modes that we erroneously kept that section in, which include
all the modes that skip finalize basically (like `WASM_BIGINT`).

This wasn't noticed before because it's just a few bytes, and because it just
happens in those specific modes (but it is annoying we didn't notice before).

### CODE REVIEW

File: emscripten.py [hunk 1]
```diff
@@ -550,6 +550,13 @@ def finalize_wasm(infile, outfile, memfile, js_syms):
   metadata = get_metadata(outfile, outfile, modify_wasm, args)
   if modify_wasm:
     building.save_intermediate(outfile, 'post_finalize.wasm')
+  else:
+    # wasm-emscripten-finalize will strip the features section for us
+    # automatically, but if we did not modify the wasm then we didn't run it,
+    # and must strip it manually. note that we must do this *after* getting the
+    # metadata, as the features are among the metadata we collect.
+    building.save_intermediate(outfile, 'strip-features.wasm')
+    building.strip(infile, outfile, sections=['target_features'])
```
Comment (tlively) [inline emscripten.py:559]: I assume this uses some other tool to very efficiently strip the section?

**Review by tlively — APPROVED** [review]
You'll have to rebaseline that metadce test, but LGTM.

File: emscripten.py (hunk 1 above)
Comment (kripken) [reply emscripten.py:559]: Yes, `llvm-objdump`

**Comment by kripken** [comment]
Rebaselined relevant tests. Mostly noise, but there is that one test that led to all this, as you said @tlively , `test_metadce_minimal_64` goes from 319 to 73 bytes.

File: emscripten.py (hunk 1 above)
Comment (sbc100) [reply emscripten.py:559]: Can we rely on `wasm-opt` to do this?  That runs in all the modes where we care about stripping it right?

Comment (sbc100) [reply emscripten.py:559]: (hopefully that will be quicker than launching an extra sub-process just for this).

Comment (kripken) [reply emscripten.py:559]: Currently we want to always strip it, even with `wasm-opt` is not run (and the metadce-minimal tests test that). But we could perhaps decide to stop caring in `-O0` and `-O1`? Let's think about that separately.

Comment (sbc100) [reply emscripten.py:559]: I think is we decide not to run wasm-opt we should probably also decide not to strip this section.  It seems like if you are building for production then wasm-opt will be run anyway, right?

**Pull request merged**